lldb/source/Host/posix/ConnectionFileDescriptorPosix.cpp

```diff
@@ -666,7 +756,23 @@ ConnectionStatus
 ConnectionFileDescriptor::ConnectFD(llvm::StringRef s,
                                     socket_id_callback_type socket_id_callback,
                                     Status *error_ptr) {
-#if LLDB_ENABLE_POSIX
+#ifdef _WIN32
+  int64_t fd = -1;
+  if (!s.getAsInteger(0, fd)) {
+    // Assume we own fd.
+    std::unique_ptr<TCPSocket> tcp_socket =
+        std::make_unique<TCPSocket>((NativeSocket)fd, true, false);
+    m_io_sp = std::move(tcp_socket);
+    m_uri = s.str();
+    return eConnectionStatusSuccess;
+  }
```

Comment:
I don't exactly have an alternative in mind yet, but I'm not sure this is a good idea. File descriptors are a thing on windows as well (though, IIUC, they're more of like an in-process thing rather than a kernel entity), and we do have code which treats them as such, so I think it could be confusing that this functions considers an "fd" to be a socket handle.

Reply:
Right, file descriptors on Windows are in-process CRT specific things. But we already have this API. I don't see any point in complicating things here.

Reply:
I do. I don't think it's a good state if half of our "fd" refer to file descriptors and others to handles. (In reality, it's not half/half, but more like 90/10, and this is the minority case/)

Reply:
What are you suggesting here? We are trying to reduce usage of `#ifdef _WIN32`. HANDLE(s) are Windows specific. We can refactor it later if necessary. I'm trying to reduce changes.

Reply:
ConnectionFileDescriptorPosix.cpp is unchanged now.
I have moved the SharedSocket using to lldb-gdbserver.cpp.
I don't like the idea about SharedSocket::GetSocket() because of ambiguity Socket/TCPSocket/UDPSocket and additional parameters should_close, child_processes_inherit, etc. It may reduce using of NativeSocket, but will not reduce the code in general.

Reply:
Okay, I think I can live with that.